src/program/lwaftr/nexthop/nexthop.lua

```diff
@@ -0,0 +1,68 @@
+module(..., package.seeall)
+
+local S = require("syscall")
+local ethernet = require("lib.protocol.ethernet")
+local lib = require("core.lib")
+local shm = require("core.shm")
+
+local ffi = require("ffi")
+local C = ffi.C
+
+local macaddress_t = ffi.typeof[[
+struct { uint8_t ether[6]; }
+]]
```

Comment:
Note for the future (doesn't need to be addressed before merging): we have a bunch of ctype pollution with various 6-byte ethernet wrappers.